### DISCUSSION TITLE
Splitting 2-up pages from a scanned book

### DISCUSSION THREAD

**Discussion by philipt18** [ImageMagick/ImageMagick#4885] [Help] [answered]
I have a PDF of a scanned book, where two pages were scanned together. I'd like to split those pages into individual pages. Assuming that it's even, what is the easiest way to split the pages in half and generate a new PDF with the split pages. For example, the original file's first page is pages 1 (left half) and 2 (right half) in the new file. I'd like to come up with a way to process the files from the command line.

Thanks.

**Comment by fmw42** [comment, marked as answer]
`convert image.pdf -crop 50x100% +repage image_%d.pdf`

will create two output images, image_0.pdf and image_1.pdf

If using IM 7, then change `convert` to `magick`.

**Reply by philipt18** [reply]
Thank you. Is there a way to take a 100 page PDF and output a 200 page PDF that is split up properly?

**Reply by fmw42** [reply]
I think you would have to write a script loop over each page of the input and then combine all them at the end.

**Reply by philipt18** [reply]
Thanks.

**Reply by fmw42** [reply]
On second thought, if you have enough RAM for the 100 page PDF at the density you need, you can do

`convert image.pdf -coalesce -crop 50x100% -adjoin new_image.pdf`

Try that on a few pages to see if it does what you want

`convert image.pdf[0-2] -coalesce -crop 50x100% -adjoin new_image.pdf`

will do the first 3 pages.

**Reply by philipt18** [reply]
That's awesome. It almost works. For some reason the pages are still 2up, but they are separated properly, they just have blank pages facing the correct ones. I'm attaching an image that shows what's going on.
![imagemagick-split-pages](https://user-images.githubusercontent.com/3592303/155494836-3e8aff34-9d85-44d4-a6f8-8b5260dad28e.jpg)
.

**Reply by fmw42** [reply]
Try adding +repage

`convert image.pdf -coalesce -crop 50x100% +repage -adjoin new_image.pdf`

**Reply by Jmuccigr** [reply]
Depending on the quality and nature of the images in the PDF, you might want to use something like pdfimages to get them out of the PDF before you play with them. Otherwise set the density before you process the images:

`magick -units pixelsperinch -density 300x300 image.pdf -coalesce -crop 50x100% +repage -adjoin new_image.pdf` 

Again, though, this will process the images, while pdfimages will just pull them out and then you can process them:

`pdfimages -all filename outputname`

**Comment by snibgo** [comment]
> I have a PDF of a scanned book, where two pages were scanned together.

So the PDF contains embedded raster images, one raster image per PDF page. If you use IM to process the PDF, it will rasterize each PDF page, which will re-sample each embedded raster image.

As Jmuccigr suggests, an alternative is to use pdfimages to extract each embedded raster image without resampling them.
```
pdfimages -all infile.pdf pref
```
This will create files named something like "pref-000.png", "pref-001.png", "pref-002.tif" etc. They have leading zeros, so they will list in the correct order. You can then use IM to write all those images to a single PDF file:
```
magick pref-* outfile.pdf
```
However, IM will attempt to read all the pref-* images into memory at the same time. If you don't have enough memory, this will cause performance problems. An alternative is to use IM to create one PDF from each abc-* file...
```
magick mogrify -format PDF abc-*
```
... then use pdfunite to glue all the resulting PDF files into a single PDF file:
```
pdfunite abc-*.PDF outfile.pdf
```

**Reply by Jmuccigr** [reply]
Another good tool is [img2pdf](https://sourceforge.net/projects/img2pdf/) which will simply insert the image into a PDF without any extra stuff, and let you choose the final page size, add margins, and so on. In my experience it can quickly do this even with hundreds of image files.

For example `img2pdf -o outputfile.pdf -S A4 image1.jpg image2.jpg`
